inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/config/DataProviderResolver.java

```diff
@@ -0,0 +1,81 @@
+import rocks.inspectit.oce.core.config.model.instrumentation.dataproviders.GenericDataProviderSettings;
+import rocks.inspectit.oce.core.instrumentation.config.model.InstrumentationConfiguration;
+import rocks.inspectit.oce.core.instrumentation.config.model.ResolvedGenericDataProviderConfig;
+import rocks.inspectit.oce.core.instrumentation.dataprovider.generic.DataProviderGenerator;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Component
+public class DataProviderResolver {
+
+    /**
+     * Resolves {@link InstrumentationConfiguration#getDataProviders()}.
+     * Returns a map mapping the names of providers to their resolved configurations
+     *
+     * @param builder the builder to resolve
+     * @param source  the input configuration
+     * @return the updated builder
+     */
+    Map<String, ResolvedGenericDataProviderConfig> resolveProviders(InstrumentationSettings source) {
+        Map<String, ResolvedGenericDataProviderConfig> resultMap = new HashMap<>();
+        for (val entry : source.getDataProviders().entrySet()) {
+            String name = entry.getKey();
+            val conf = entry.getValue();
+
+            HashMap<String, String> additionalInputs = new HashMap<>(conf.getInput());
+
+            val result = ResolvedGenericDataProviderConfig.builder()
+                    .name(name)
+                    .importedPackages(conf.getImports());
+
+            resolveSpecialVariables(additionalInputs, result);
+            resolveArgumentVariables(additionalInputs, result);
+            resolveAdditionalInputs(additionalInputs, result);
+            resolveBody(conf, result);
+
+            resultMap.put(name, result.build());
+        }
+        return resultMap;
+    }
+
+    private void resolveBody(GenericDataProviderSettings conf, ResolvedGenericDataProviderConfig.ResolvedGenericDataProviderConfigBuilder result) {
+        if (conf.getValue() != null) {
+            result.valueBody("return " + conf.getValue() + ";");
+        } else {
+            result.valueBody(conf.getValueBody());
+        }
+    }
+
+    private void resolveAdditionalInputs(HashMap<String, String> additionalInputs, ResolvedGenericDataProviderConfig.ResolvedGenericDataProviderConfigBuilder result) {
+        //everything that has not been removed is an additional argument
+        for (val additionalVarEntry : additionalInputs.entrySet()) {
+            result.additionalArgumentType(additionalVarEntry.getKey(), additionalVarEntry.getValue());
+        }
+    }
+
+    private void resolveArgumentVariables(HashMap<String, String> inputs, ResolvedGenericDataProviderConfig.ResolvedGenericDataProviderConfigBuilder result) {
+        val entryIterator = inputs.entrySet().iterator();
+        while (entryIterator.hasNext()) {
+            val varEntry = entryIterator.next();
+            String varName = varEntry.getKey();
+            if (varName.matches(DataProviderGenerator.ARG_VARIABLE_PREFIX + "\\d+")) {
```

Comment:
not sure if we should use the greedy variant `\\d*` which will only match if rest of the string are numbers..

this way you would match: `arg1 and_antyhing_else ` and you would then fail with the `parseInt`.

Reply:
String.matches matches the entire string, not only a prefix. This means for your example it  would not match the pattern.